rfcs/text/0008-email.md

```diff
@@ -0,0 +1,155 @@
+# 0008: Email
+<!-- Leave this ID at 0000. The ECS team will assign a unique, contiguous RFC number upon merging the initial stage of this RFC. -->
+
+- Stage: **1 (proposal)** <!-- Update to reflect target stage. See https://elastic.github.io/ecs/stages.html -->
+- Date: **Oct 5th 2020** <!-- The ECS team sets this date at merge time. This is the date of the latest stage advancement. -->
+
+This RFC proposes a new top-level field to facilitate email use cases.
+
+<!--
+As you work on your RFC, use the "Stage N" comments to guide you in what you should focus on, for the stage you're targeting.
+Feel free to remove these comments as you go along.
+-->
+
+<!--
+Stage 0: Provide a high level summary of the premise of these changes. Briefly describe the nature, purpose, and impact of the changes. ~2-5 sentences.
+-->
+
+## Fields
+
+<!--
+Stage 1: Describe at a high level how this change affects fields. Which fieldsets will be impacted? How many fields overall? Are we primarily adding fields, removing fields, or changing existing fields? The goal here is to understand the fundamental technical implications and likely extent of these changes. ~2-5 sentences.
+-->
+
+Email specific fields:
+
+| field | type | description |
+| --- | --- | --- |
+| `email.bcc.addresses` | wildcard | Addresses of Bcc's |
+| `email.cc.addresses` | wildcard | Addresses of Cc's |
+| `email.attachments_count` | long | A field outside the flattened structure to control how many attachments are included in the email |
+| `email.attachments` | flattened | A flattened field for anything related to attachments. This allows objects being stored with all information for each file when you have multiple attachments |
+| `email.direction` | keyword | Direction of the message based on the sending and receving domains |
+| `email.sender.address` | wildcard | Senders email address |
+| `email.sender.top_level_domain` | keyword | Senders email address |
```

Comment:
Some additional fields that might be useful for the emails.sender fields:
- email.sender.domain | keyword | Domain of the sender
- email.sender.registered_domain | keyword | Registered domain of the sender (this is useful when dealing with reporting that involves multiple sub-domains sending emails)

These fields are useful for pivoting across large amounts of emails, as well as in aggregations for reporting. (email.sender.domain currently falls under email.domains, but capturing direction in reports is not possible via a related field)

Reply:
Thanks @BenB196 - Good point to note, and thanks for the feedback about the fields being useful for pivoting and aggs.

There was some work done recently (https://github.com/elastic/ecs/pull/981) to ensure the domain name breakdown fields (`*.domain` + `*.registered_domain`, `top_level_domain`, and `*.subdomain`) were consistent across field sets. I expect we'll want to apply that same uniformity here.

Reply:
I agree with both. This is something we had been discussing as well for email, though the discussion was currently around top_level_domain only.
Since sender is a single address, I do see the reasons why we should have maybe a separate domain field (outside of `email.domains`).

I will add in some additions based on the comment from @ebeahan as well, and we could use that as a basis for any upcoming discussions as it does not need to be set in stone just yet.